FirebaseRemoteConfig/Sources/RCNConfigRealtime.m

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import "FirebaseRemoteConfig/Sources/RCNConfigRealtime.h"
+#import <Foundation/Foundation.h>
+#import "FirebaseRemoteConfig/Sources/RCNConfigConstants.h"
+
+@implementation FIRConfigUpdateListenerRegistration {
+  RCNConfigRealtime *_realtimeClient;
+  NSString *_listenerName;
+}
+
+- (instancetype)initWithClient:(RCNConfigRealtime *)realtimeClient
+                  listenerName:(NSString *)listenerName {
+  self = [super init];
+  if (self) {
+    _realtimeClient = realtimeClient;
+    _listenerName = listenerName;
+  }
+  return self;
+}
+
+- (void)remove {
+  [self->_realtimeClient removeConfigUpdateListener:_listenerName];
+}
+
+@end
+
+@interface RCNConfigRealtime ()
+
+@property(strong, atomic, nonnull) NSMutableDictionary<NSString *, id> *listeners;
+@property(strong, atomic, nonnull) dispatch_queue_t lockQueue;
+@property(nonatomic, assign) NSInteger listenerCount;
+
+@end
+
+@implementation RCNConfigRealtime
+
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    _listeners = [NSMutableDictionary alloc];
+    _lockQueue = [RCNConfigRealtime realtimeRemoteConfigSerialQueue];
+    _listenerCount = 1;
+  }
+
+  return self;
+}
+
+/// Singleton instance of serial queue for queuing all incoming RC calls.
++ (dispatch_queue_t)realtimeRemoteConfigSerialQueue {
+  static dispatch_once_t onceToken;
+  static dispatch_queue_t realtimeRemoteConfigQueue;
+  dispatch_once(&onceToken, ^{
+    realtimeRemoteConfigQueue =
+        dispatch_queue_create(RCNRemoteConfigQueueLabel, DISPATCH_QUEUE_SERIAL);
+  });
+  return realtimeRemoteConfigQueue;
+}
+
+- (void)beginRealtimeStream {
+}
+
+- (void)pauseRealtimeStream {
+}
+
+- (FIRConfigUpdateListenerRegistration *)addConfigUpdateListener:
+    (FIRConfigUpdateCompletion)completionHandler {
+  NSString *listenerName = [@(self->_listenerCount) stringValue];
+  dispatch_async(_lockQueue, ^{
+    [self->_listeners setValue:completionHandler forKey:listenerName];
+    self->_listenerCount++;
+  });
+
+  [self beginRealtimeStream];
+  return [[FIRConfigUpdateListenerRegistration alloc] initWithClient:self
+                                                        listenerName:listenerName];
+}
+
+- (void)removeConfigUpdateListener:(NSString *)listenerName {
+  dispatch_async(_lockQueue, ^{
+    [self->_listeners removeObjectForKey:listenerName];
+  });
+  if ([self->_listeners count] == 0) {
```

Comment:
nit use .count property is preferred for obcC

Reply:
Done, I'll add unit tests in the next PR when I have the HTTP stream logic as well.